File: damon/src/main/java/com/zyhang/damon/RequiresPresenter.java
package com.zyhang.damon;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * ProjectName:Damon
 * Description:
 * Created by zyhang on 2017/4/28.下午10:58
 * Modify by:
 * Modify time:
 * Modify remark:
 */

@Inherited
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface RequiresPresenter {
    Class<? extends MvpPresenter> value();
}

File: backup/src/main/kotlin/com/zyhang/damon/kotlin/RequiresPresenter.kt
package com.zyhang.damon.kotlin

import java.lang.annotation.Inherited
import kotlin.reflect.KClass

/**
 * ProjectName:Damon
 * Description:
 * Created by zyhang on 2017/4/28.下午5:52
 * Modify by:
 * Modify time:
 * Modify remark:
 */

@Inherited
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class RequiresPresenter(val cls: KClass<out MvpPresenter<MvpView>>)